### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by nik2143** [ByNameModding/BNM-Android#9]
The last release of bnm crashes without a reason

**Comment by BNM-Dev** [comment]
Yoy can show crash log?

**Comment by nik2143** [comment]
No crash log in logcat it only force close

**Comment by BNM-Dev** [comment]
Any logs by ByNameModding tag?

**Comment by nik2143** [comment]
No, 0 logs from BNM tag

**Comment by BNM-Dev** [comment]
Uncomment `BNM_DISABLE_AUTO_LOAD` in BNM_settings.h and call `BNM::HardBypass` in `JNI_OnLoad`

**Comment by nik2143** [comment]
`2023-07-23 01:55:38.581  7544-7630  ByNameModding           com.kiloo.subwaysurf                 D  [InitIl2cppMethods] il2cpp::vm::Class::Init in lib: 0x57baa0
2023-07-23 01:55:38.581  7544-7630  ByNameModding           com.kiloo.subwaysurf                 D  [InitIl2cppMethods] game has il2cpp_image_get_class. BNM will use it
2023-07-23 01:55:38.581  7544-7630  ByNameModding           com.kiloo.subwaysurf                 D  [InitIl2cppMethods] il2cpp::vm::Class::FromIl2CppType in lib: 0x57e564
2023-07-23 01:55:38.581  7544-7630  ByNameModding           com.kiloo.subwaysurf                 D  [InitIl2cppMethods] il2cpp::vm::Type::GetClassOrElementClass в библиотеке: 0x59ca0c.
2023-07-23 01:55:38.581  7544-7630  ByNameModding           com.kiloo.subwaysurf                 D  [InitIl2cppMethods] il2cpp::vm::Image::FromName in lib: 0x582780
2023-07-23 01:55:38.581  7544-7630  ByNameModding           com.kiloo.subwaysurf                 D  [InitIl2cppMethods] il2cpp::vm::Assembly::GetAllAssemblies by domain in lib: 0x57bc84`

**Comment by Scriptor-Lover1500** [comment]
@nik2143 I am also facing the same crash in latest bnm. Have you already fixed it or still crashing?

**Comment by nik2143** [comment]
> @nik2143 I am also facing the same crash in latest bnm. Have you already fixed it or still crashing?

Still crashing

**Comment by BNM-Dev** [comment]
What game? I'll check it myself when I have time

**Comment by nik2143** [comment]
Subway surfers

**Comment by BNM-Dev** [comment]
Add this to code (to JNI on load or other methods that starts with lib load):
```
void il2cpp_thread() {

}

    // ...
    BNM::AddOnLoadedEvent(il2cpp_thread);
    // ...
 ```

**Comment by nik2143** [comment]
It still crash

**Comment by PastMC** [comment]
try change #define BNM_DISABLE_NEW_CLASSES 0 to #define BNM_DISABLE_NEW_CLASSES 1

**Comment by nik2143** [comment]
Done , not crash but stucks on black screen

**Comment by PastMC** [comment]
try make with std::thread(hack_thread).detach(); load and in bnm.cpp BNM_il2cpp_init remove all what use events

**Comment by PastMC** [comment]
and if you not use BNM::HardBypass use it

**Comment by nik2143** [comment]
Thanks, this solved the issue

**Comment by PastMC** [comment]
close issue if no any problems
